x-pack/plugins/custom_branding/server/plugin.ts

```diff
@@ -48,14 +52,25 @@ export class CustomBrandingPlugin implements Plugin {
     const router = core.http.createRouter<CustomBrandingRequestHandlerContext>();
     registerRoutes(router);
 
-    const fetchFn = async (request: KibanaRequest): Promise<CustomBranding> => {
+    const fetchFn = async (
+      request: KibanaRequest,
+      unauthenticated?: boolean
+    ): Promise<CustomBranding> => {
+      if (!this.isValidLicense) {
+        return {};
+      }
       const [coreStart] = await core.getStartServices();
-      const soClient = coreStart.savedObjects.getScopedClient(request);
+      const soClient = unauthenticated
+        ? coreStart.savedObjects.getScopedClient(request, {
+            excludedExtensions: [SECURITY_EXTENSION_ID],
+          })
+        : coreStart.savedObjects.getScopedClient(request);
```

Comment:
Probably just a NIT, but: 

Why exactly do we need/want to propagate the `unauthenticated` information here? What are the upsides compared to always create the SO client with  `excludedExtensions: [SECURITY_EXTENSION_ID]` in both cases (auth or not auth?)

Reply:
I thought excluding security extension was bad practice and we should always use it unless absolutely necessary not to?

Reply:
Yeah, you're probably right. It's just that having a more strict security check for authenticated user than unauthenticated ones feels... awkward 😅 .